crypto/fipsmodule/service_indicator/service_indicator_test.cc

```diff
@@ -259,6 +259,35 @@ static const uint8_t kAESCMACOutput[16] = {0xe7, 0x32, 0x43, 0xb4, 0xae, 0x79,
                                            0x08, 0x86, 0xe7, 0x9f, 0x0d, 0x3f,
                                            0x88, 0x3f, 0x1a, 0xfd};
 
+// AES-XTS test vector from
+// https://csrc.nist.gov/projects/cryptographic-algorithm-validation-program
+
+static const uint8_t kAESXTSKey_256[64] = {
+    0x1e, 0xa6, 0x61, 0xc5, 0x8d, 0x94, 0x3a, 0x0e, 0x48, 0x01, 0xe4, 0x2f,
+    0x4b, 0x09, 0x47, 0x14, 0x9e, 0x7f, 0x9f, 0x8e, 0x3e, 0x68, 0xd0, 0xc7,
+    0x50, 0x52, 0x10, 0xbd, 0x31, 0x1a, 0x0e, 0x7c, 0xd6, 0xe1, 0x3f, 0xfd,
+    0xf2, 0x41, 0x8d, 0x8d, 0x19, 0x11, 0xc0, 0x04, 0xcd, 0xa5, 0x8d, 0xa3,
+    0xd6, 0x19, 0xb7, 0xe2, 0xb9, 0x14, 0x1e, 0x58, 0x31, 0x8e, 0xea, 0x39,
+    0x2c, 0xf4, 0x1b, 0x08
+};
+
+static const uint8_t kAESXTSIV_256[16] = {
+    0xad, 0xf8, 0xd9, 0x26, 0x27, 0x46, 0x4a, 0xd2, 0xf0, 0x42, 0x8e, 0x84,
+    0xa9, 0xf8, 0x75, 0x64
+};
+
+static const uint8_t kAESXTSPlaintext_256[32] = {
```

Comment:
I'm not too familiar with the semantics of `XTS`, but is AES-XTS only able to take in 32 bit test lengths?

Reply:
I also have no idea, I'm just moving code around. I'll look at [NIST SP800-38e](https://csrc.nist.gov/publications/detail/sp/800-38e/final) and see if I can find out.

Reply:
XTS mode operates on fixed size blocks (<= 2^20^ AES blocks); this size doesn't need to be a multiple of the AES block size, the incomplete block at the end will be padded out with ciphertext stealing. The output ciphertext will be the same size as the input plaintext.

Keys must be 128 or 256 bits. I'll add some additional test cases.

Reply:
Looks like the AWS-LC implementation requires 256-bit keys and a multiple of the AES block size for the data (tested with 256, 384, 140, 250, 384-bit data from the NIST KATs).

Reply:
Thanks for all the details, that was really enlightening.